include/nbl/asset/IDescriptorSet.h

```diff
@@ -47,9 +47,10 @@ class IDescriptorSet : public virtual core::IReferenceCounted // TODO: try to re
                 };
                 struct SImageInfo
                 {
-					// This will be ignored if the DS layout already has an immutable sampler specified for the binding.
-                    core::smart_refctd_ptr<typename layout_t::sampler_type> sampler;
-                    IImage::LAYOUT imageLayout;
+					IImage::LAYOUT imageLayout;
+					// If the binding is COMBINED, this will be ignored if the DS layout already has an immutable sampler specified for the binding
+					// If the binding is an immutable SAMPLER, whatever's here doesn't matter because the write validation will fail
```

Comment:
if the binding is `SAMPLER` then its the `desc` thats the sampler, not this `sampler`

Reply:
this variable is only relevant for `COMBINED` and Mutable bindings, otherwise gets ignored or fails validation

Reply:
`imageLayout` ofc not ignored for `SAMPLED_IMAGE`